crypt.h.in.in

```diff
@@ -178,12 +178,13 @@ extern char *crypt_gensalt_rn (const char *__prefix, unsigned long __count,
                                char *__output, int __output_size)
 __THROW;
 
-/* Identical to crypt_gensalt_rn.  Kept for code compatibility with
-   older versions (v3.1.1 and earlier) of libxcrypt.  */
-extern char *crypt_gensalt_r (const char *__prefix, unsigned long __count,
-                              const char *__rbytes, int __nrbytes,
-                              char *__output, int __output_size)
-__THROW;
+/* Kept for code compatibility with libxcrypt (v3.1.1 and earlier).
+   We intentionally define a macro here, since we actually want to
+   link compiled applications against the identical crypt_gensalt_rn
+   function.  */
+#ifndef IN_LIBCRYPT  /* Defined when building libxcrypt. */
+# define crypt_gensalt_r crypt_gensalt_rn
+#endif
```

Comment:
It would be better to do this with GCC's `__attribute__ ((alias ("crypt_gensalt_rn")))` when possible. glibc's sys/cdefs.h defines a macro called `__REDIRECT_NTH` that encapsulates "when possible" and we're already including that header so let's use that.

Reply:
We still need keep `#ifndef IN_LIBCRYPT`, tho, before declaring the fuctions using `__REDIRECT_NTH`, because we need to create a `strong_alias` when building libxcrypt.

Anyways, I've introduced to prefer `__REDIRECT_NTH`, when available.